### DISCUSSION TITLE
Economy API

### DISCUSSION THREAD

**Discussion by coderFlameyosFlow** [BG-Software-LLC/SuperiorSkyblock2#1611] [Pending Feature Requests] [answered]
API methods for SuperiorSkyblock that helps implement your own Economy for SuperiorSkyblock or let an economy plugin support SuperiorSkyblock (and Vault could be a good way to do that, thanks to its popularity).

Really nice choice for people that want more control over their server's Economy rather than only using EssentialsX.

This is because EssentialsX *could lack more features and performance*, 

dedicated economy plugins usually are more performant and feature-rich because

(performance-wise): they have the option **to use an SQL based database instead of some files and can have async processes**

(feature-wise): they are able to 
- set more options 
- have more databases to work with
- a little more readable *(because there aren't 2 million features crunched in one plugin)*
- can include cool system features like interest, trading, minecraft:paper money and more**
- more features plugin-wise

All thanks to them being dedicated economy plugins.

**Comment by OmerBenGera** [comment, marked as answer]
https://bg-software.com/superiorskyblock/api/
Look for the EconomyProvider class - this is used by the plugin to interact with economy plugins.
You can register your custom economy provider from the ProvidersManager class.
